readthedocs/builds/version_slug.py

```diff
@@ -50,36 +48,23 @@ def get_fields_with_model(cls):
 #   (?: ... ) -- wrap everything so that the pattern cannot escape when used in
 #                regexes.
 VERSION_SLUG_REGEX = '(?:[a-z0-9A-Z][-._a-z0-9A-Z]*?)'
+VERSION_OK_CHARS = '-._'  # dash, dot, underscore
+VERSION_TEST_PATTERN = re.compile('^{pattern}$'.format(pattern=VERSION_SLUG_REGEX))
+VERSION_FALLBACK_SLUG = 'unknown'
 
 
-class VersionSlugField(models.CharField):
+class VersionSlugify:
```

Comment:
I understand that this new class is just a refactor to get out some hardcoded code from the VersionSlugField to be able to use it from the Form validation as well. Am I right?

In case that you change some logic, can you pointed to me? I didn't really find any.

Reply:
Yeah, it just extracts the code to make it reusable